### PR TITLE
Upgraded contract version to contain the Diagnose contract

### PR DESCRIPTION
**Pull request by yanivid3**
Upgraded contract version to contain the Diagnose contract

### CODE REVIEW

File: src/SDK/SmartDetectorsShared/Package/SmartDetectorManifest.cs [hunk 1]
```diff
@@ -8,8 +8,7 @@ namespace Microsoft.Azure.Monitoring.SmartDetectors.Package
 {
     using System;
     using System.Collections.Generic;
-    using System.Linq;
-    using Microsoft.Azure.Monitoring.SmartDetectors.RuntimeEnvironment.Contracts;
+        using Microsoft.Azure.Monitoring.SmartDetectors.RuntimeEnvironment.Contracts;
```
Comment (liadtal) [inline src/SDK/SmartDetectorsShared/Package/SmartDetectorManifest.cs]: >     [](start = 4, length = 4)

While you're at it - redundant space...

File: src/SDK/SmartDetectorsShared/Clients/TelemetryDbType.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+namespace Microsoft.Azure.Monitoring.SmartDetectors.RuntimeEnvironment.Contracts
+{
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Converters;
+
+    /// <summary>
+    /// An enum for the type of DB on which to run queries that are part of alerts presentation
+    /// </summary>
+    [JsonConverter(typeof(StringEnumConverter))]
+    public enum TelemetryDbType
```
Comment (EranGon) [inline src/SDK/SmartDetectorsShared/Clients/TelemetryDbType.cs]: >TelemetryDbType [](start = 16, length = 15)

huh? please remove

Comment (EranGon) [reply src/SDK/SmartDetectorsShared/Clients/TelemetryDbType.cs]: pass a string to the TelemetryDbType class ctor and use it for telemetry

---
In reply to: [257923834](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/49#discussion_r257923834) [](ancestors = 257923834)

File: src/SDK/SmartDetectors.Extensions/SmartDetectors.Extensions.csproj [hunk 3]
```diff
@@ -9,7 +9,7 @@
     <GenerateDocumentationFile>true</GenerateDocumentationFile>
   </PropertyGroup>  
   <ItemGroup>
-    <PackageReference Include="Microsoft.Azure.Monitoring.SmartDetectors.RuntimeEnvironment.Contracts" Version="0.16.0" />
+    <PackageReference Include="Microsoft.Azure.Monitoring.SmartDetectors.RuntimeEnvironment.Contracts" Version="1.0.9.177" />
```
Comment (EranGon) [inline src/SDK/SmartDetectors.Extensions/SmartDetectors.Extensions.csproj]: >1.0.9.177 [](start = 112, length = 9)

don't you have to update Nuget.config?

Comment (yanivid3) [reply src/SDK/SmartDetectors.Extensions/SmartDetectors.Extensions.csproj]: I updated

---
In reply to: [257946149](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/49#discussion_r257946149) [](ancestors = 257946149)

File: src/SDK/MonitoringApplianceEmulator/Controls/KeyValuePropertyControl.xaml.cs [hunk 4]
```diff
@@ -8,7 +8,7 @@ namespace Microsoft.Azure.Monitoring.SmartDetectors.MonitoringApplianceEmulator.
 {
     using System.Windows;
     using System.Windows.Controls;
-    using RuntimeEnvironment.Contracts;
+    using RuntimeEnvironment.Contracts.AlertProperties;
```
Comment (EranGon) [inline src/SDK/MonitoringApplianceEmulator/Controls/KeyValuePropertyControl.xaml.cs]: >RuntimeEnvironment.Contracts.AlertProperties [](start = 10, length = 44)

please change this to be the full namespace

File: src/SDK/MonitoringApplianceEmulator/Controls/TextPropertyControl.xaml.cs [hunk 5]
```diff
@@ -8,7 +8,7 @@ namespace Microsoft.Azure.Monitoring.SmartDetectors.MonitoringApplianceEmulator.
 {
     using System.Windows;
     using System.Windows.Controls;
-    using RuntimeEnvironment.Contracts;
+    using RuntimeEnvironment.Contracts.AlertProperties;
```
Comment (EranGon) [inline src/SDK/MonitoringApplianceEmulator/Controls/TextPropertyControl.xaml.cs]: full namespace

File: src/SDK/SmartDetectors.Extensions/SmartDetectors.Extensions.csproj (hunk 3 above)
Comment (EranGon) [reply src/SDK/SmartDetectors.Extensions/SmartDetectors.Extensions.csproj]: yes - add the following line to nuget.config:
<add key="OneBranchOfficial" value="[https://msazure.pkgs.visualstudio.com/DefaultCollection/_apis/packaging/Official/nuget/index.json](https://msazure.pkgs.visualstudio.com/DefaultCollection/_apis/packaging/Official/nuget/index.json)" />

---
In reply to: [258010304](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/49#discussion_r258010304) [](ancestors = 258010304,257946149)

File: src/SDK/SmartDetectorsShared/Clients/TelemetryDataClientBase.cs [hunk 6]
```diff
@@ -86,7 +86,7 @@ public abstract class TelemetryDataClientBase : ITelemetryDataClient
         /// <summary>
         /// Gets the type of telemetry DB that this data client accesses.
         /// </summary>
-        public TelemetryDbType TelemetryDbType { get; }
+        public string TelemetryDbType { get; }
```
Comment (EranGon) [inline src/SDK/SmartDetectorsShared/Clients/TelemetryDataClientBase.cs]: >public string TelemetryDbType { get; } [](start = 8, length = 38)

make this private readonly member

**Comment by yanivid3** [comment]
Thanks

**Pull request merged**